### PR TITLE
Add vertical pod autoscaler

### PR DESCRIPTION
**Pull request by yulianedyalkova**
Towards https://github.com/giantswarm/giantswarm/issues/12395

This will create a vpa for each kvm-operator deployment. Testing in on gorgoth I got:
```
Recommendation:
  Container Recommendations:
    Container Name:  kvm-operator
    Lower Bound:
      Cpu:     33m
      Memory:  104857600
    Target:
      Cpu:     63m
      Memory:  272061154
    Uncapped Target:
      Cpu:     63m
      Memory:  272061154
    Upper Bound:
      Cpu:     2309m
      Memory:  4986478263
```

where: 
* Target: the requests values that will get configured on the pods at the next iteration.
* Lower/Upper bound: the values that the pod needs to go under/above to get **evicted and up/downscaled**.

The limits to request ratio gets preserved from the values passed in the deployment initially. So in our case they will be [1:1](https://github.com/giantswarm/kvm-operator/blob/12ebe236628d4360ce2ce6b466c612c179322181/helm/kvm-operator/templates/deployment.yaml#L58-L64). 

Also the more it runs, the more data about the usage it gets so the predictions get better. After about an hour of running, the recommendation changed to (upper bound wasn't that random anymore):

```
 Recommendation:
   Container Recommendations:
     Container Name:  kvm-operator
     Lower Bound:
       Cpu:     34m
       Memory:  104857600
     Target:
       Cpu:     49m
       Memory:  272061154
     Uncapped Target:
       Cpu:     49m
       Memory:  272061154
     Upper Bound:
       Cpu:     879m
       Memory:  3798855408
```

Logs in the `vertical-pod-autoscaler` to prove that it actually does something:
```
I0105 08:45:19.416775       1 update_priority_calculator.go:133] not updating a short-lived pod giantswarm/kvm-operator-vpa-75fcddc7d7-dxrrb, request within recommended range
I0105 08:46:19.419168       1 update_priority_calculator.go:133] not updating a short-lived pod giantswarm/kvm-operator-vpa-75fcddc7d7-dxrrb, request within recommended range
I0105 08:47:19.417194       1 update_priority_calculator.go:147] pod accepted for update giantswarm/kvm-operator-vpa-75fcddc7d7-dxrrb with priority 0.7858309402465821
I0105 08:47:19.417273       1 updater.go:193] evicting pod kvm-operator-vpa-75fcddc7d7-dxrrb
I0105 08:48:19.418728       1 update_priority_calculator.go:133] not updating a short-lived pod giantswarm/kvm-operator-vpa-75fcddc7d7-2gbwq, request within recommended range
I0105 08:49:19.423243       1 update_priority_calculator.go:133] not updating a short-lived pod giantswarm/kvm-operator-vpa-75fcddc7d7-2gbwq, request within recommended range
```

### CODE REVIEW

File: helm/kvm-operator/templates/vpa.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  namespace: {{ include "resource.default.namespace" . }}
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+spec:
+  resourcePolicy:
+    containerPolicies:
+    - containerName: {{ .Chart.Name }}
+      controlledValues: RequestsAndLimits
+      mode: Auto
+  targetRef:
```
Comment (MarcelMue) [inline helm/kvm-operator/templates/vpa.yaml:14]: Does this ref not include a namespace?

**Review by MarcelMue — APPROVED** [review]
Seems reasonable to me

File: helm/kvm-operator/templates/vpa.yaml (hunk 1 above)
Comment (yulianedyalkova) [reply helm/kvm-operator/templates/vpa.yaml:14]: No it doesn't ([link](https://v1-16.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#crossversionobjectreference-v1-autoscaling)). I'm assuming it just looks for resources in the namespace where it's running.

**Comment by yulianedyalkova** [comment]
I've also set an upper limit to the requests and limits that's twice as big as the values we had so far.

**Pull request merged**